### PR TITLE
Make evalml compatible with `woodwork == 0.20.0`, upgrade to `0.21.0`

### PR DESCRIPTION
**Pull request by ParthivNaresh**

### CODE REVIEW

**Comment by ParthivNaresh** [comment]
Windows tests are failing here due to the editable file installation of woodwork main I believe. This should resolve itself once Woodwork 0.20.0 is released.

**Comment by ParthivNaresh** [comment]
@chukarsten I might need you to force merge this for me once approved. There's one line not covered which is the `TypeError` check for the inference function. It doesn't need to be tested since it's an edge case logic that's already tested in Woodwork.

File: .github/meta.yaml
Comment (eccabay) [inline .github/meta.yaml:40]: Are the changes in this PR incompatible with older versions of woodwork?

File: docs/source/release_notes.rst
Comment (eccabay) [inline docs/source/release_notes.rst:10]: Can you add a line about upgrading the woodwork version as well? (only really useful since there's the note about pinning it in here)

File: evalml/utils/woodwork_utils.py
Comment (eccabay) [inline evalml/utils/woodwork_utils.py:54]: This is wacky. If I understand correctly, we're directly changing woodwork inference behavior during init of the evalml package? Why is this necessary for evalml? It seems odd to fork off of standard woodwork behavior, it may cause headaches in the future.

File: evalml/utils/woodwork_utils.py
Comment (eccabay) [inline evalml/utils/woodwork_utils.py:31]: Even though this function is private, context on what it's doing would be useful for future work. Could you add a docstring?

**Review by eccabay — COMMENTED** [review]
Thanks for taking care of this Parthiv! I just have a few questions before we go ahead and merge this in

File: evalml/utils/woodwork_utils.py
Comment (ParthivNaresh) [reply evalml/utils/woodwork_utils.py:54]: This was the reason that the `update_inference_function` was added in Woodwork, so that inference functions could be changed/added whenever necessary.

Unfortunately the alternative is to not only update every test that runs into this Boolean inference issue, but also to update several components like the `LabelEncoder` that run into reverse mapping issues. In order for the new Boolean inference to be supported here, there needs to be an underlying change in cases where previously `Categorical` variables will be treated as `Boolean` instead.

Luckily any downstream `Boolean` transformations have to go through this inference first, so this can act as a sort of "gatekeeper" for this behaviour, for now.

Comment (jeremyliweishih) [reply evalml/utils/woodwork_utils.py:54]: @ParthivNaresh is what EvalML considers "boolean" different from what WW does now? and what is the difference? Just curious why inference shouldn't be standard between WW and EvalML.

File: .github/meta.yaml
Comment (ParthivNaresh) [reply .github/meta.yaml]: Confirmed that `0.19.0` works fine while `0.20.0` has dependency issues. I updated the dependencies appropriately!

**Review by jeremyliweishih — APPROVED** [review]
LGTM!

**Review by eccabay — APPROVED** [review]

**Review by chukarsten — CHANGES_REQUESTED** [review]
I've got a bad feeling about this

**Pull request closed without merging**
